UIforETW/DirectoryMonitor.cpp

```diff
@@ -62,21 +61,23 @@ DWORD DirectoryMonitor::DirectoryMonitorThread()
 			return 0;
 
 		default:
-			assert(0);
+			ATLASSERT(0);
 			break;
 		}
 	}
   // Unreachable.
 }
 
+_Pre_satisfies_(this->hThread_ == 0)

```

Comment:
Yeah... but still, ugly. Especially the "_In_ DWORD ThreadId" which conveys zero information.